frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx

```diff
@@ -0,0 +1,332 @@
+  }, [dispatch, list, entityDefinition, wrapped]);
+
+  const reloadInterval = useSelector(state => {
+    if (typeof reloadIntervalProp === "function") {
+      return reloadIntervalProp(state, props, list);
+    }
+    return reloadIntervalProp;
+  });
+
+  const {
+    data,
+    error: rtkError,
+    isFetching,
+    isLoading,
+    refetch,
+  } = entityDefinition.rtk.useListQuery(entityQuery, {
+    pollingInterval: reloadInterval,
+    refetchOnMountOrArgChange: reload,
+  });
+
+  const queryKey = useMemo(
+    () => entityDefinition.getQueryKey(entityQuery),
+    [entityDefinition, entityQuery],
+  );
+
+  const listStatePath = useMemo(() => {
+    return entityDefinition.getListStatePath(entityQuery);
+  }, [entityDefinition, entityQuery]);
+
+  const requestStatePath = useMemo(() => {
+    return [...listStatePath, "fetch"];
+  }, [listStatePath]);
+
+  useEffect(() => {
+    if (isFetching) {
+      // @ts-expect-error - invalid typings in redux-actions package
+      dispatch(setRequestLoading(requestStatePath, queryKey));
+    }
+  }, [dispatch, isFetching, requestStatePath, queryKey]);
+
+  useEffect(() => {
+    if (rtkError) {
+      // @ts-expect-error - invalid typings in redux-actions package
+      dispatch(setRequestError(requestStatePath, queryKey, rtkError));
+    }
+  }, [dispatch, rtkError, requestStatePath, queryKey]);
+
+  useEffect(() => {
+    if (data) {
+      const { results, metadata } = transformResponse(data);
+
+      if (!Array.isArray(results)) {
+        throw new Error(`Invalid response listing ${entityDefinition.name}`);
+      }
+
+      const normalized = entityDefinition.normalizeList(results);
+
+      const result = {
+        ...normalized,
+        metadata,
+        entityQuery,
+      };
+
+      dispatch({
+        type: entityDefinition.actionTypes.FETCH_LIST,
+        payload: result,
+      });
+
+      // NOTE Atte Keinänen 8/23/17:
+      // Dispatch `setRequestLoaded` after clearing the call stack because we want to the actual data to be updated
+      // before we notify components via `state.requests.fetches` that fetching the data is completed
+      // @ts-expect-error - invalid typings in redux-actions package
+      setTimeout(() => dispatch(setRequestLoaded(requestStatePath, queryKey)));
+    }
+  }, [
+    dispatch,
+    data,
+    entityDefinition,
+    entityQuery,
+    requestStatePath,
+    queryKey,
+  ]);
+
+  const onLoadedRef = useLatest(onLoaded);
+
+  useEffect(() => {
+    if (data) {
+      onLoadedRef.current?.(data);
+    }
+  }, [data, onLoadedRef]);
+
+  // merge props passed in from stacked Entity*Loaders:
+  const allError = error || (allErrorProp == null ? null : allErrorProp);
+  const allFetched =
+    fetched && (allFetchedProp == null ? true : allFetchedProp);
+  const allLoaded = loaded && (allLoadedProp == null ? true : allLoadedProp);
+  const allLoading =
+    loading || (allLoadingProp == null ? false : allLoadingProp);
```

Comment:
Not sure what the exact semantics are here, but I think this is clearer:

```suggestion
  const allError = error || (allErrorProp ?? null);
  const allFetched = fetched && (allFetchedProp ?? true);
  const allLoaded = loaded && (allLoadedProp ?? true);
  const allLoading = loading || (allLoadingProp ?? false);
```

Reply:
0118847ed8fe95705a88f617f048de2359970a59